File: techniques/JS_PYServer/req.js
/***
Javier Sorribes, Hispasat S.A.
06/07/2017

Requests to TotalNMS' NBI

Requires ../helpers.js
***/
var host = 'http://127.0.0.1:9000';
var username = 'admin';
var password = 'manager';

function getCPEsByManagedGroup(mgid) {
	document.getElementById('numCPEs').innerHTML = 'Loading...'
	var xhttp = new XMLHttpRequest();
	xhttp.onreadystatechange = function() {
		if (this.readyState == 4) {
            if (this.status == 200) {
			    xmlDoc = xhttp.responseXML;
			    ret = xmlDoc.childNodes[0].childNodes[0].childNodes[0];
			    document.getElementById('numCPEs').innerHTML = ret.getElementsByTagName('totalNumber')[0].childNodes[0].nodeValue;
            }
		    else if (this.status == 500) {
                xmlDoc = xhttp.responseXML;
                fault = xmlDoc.childNodes[0].childNodes[0];
			    document.getElementById('numCPEs').innerHTML = fault.getElementsByTagName('faultstring')[0].childNodes[0].nodeValue;
            }
            else {
                document.getElementById('numCPEs').innerHTML = 'Not successful, status code: ' + this.status;
            }
		}
	};
	xhttp.open('POST', host, true);//, username, password);
	//xhttp.withCredentials = true;
    xhttp.setRequestHeader('Authorization', 'Basic ' + btoa(username+':'+password));
	xhttp.setRequestHeader('Content-Type', 'text/xml;charset=UTF-8');
	xhttp.setRequestHeader('SOAPAction', '');
	//xhttp.setRequestHeader('Connection', 'keep-alive');  //not required
	var req;
	readXMLFile('/src/XMLReqs/getCPEsByManagedGroup.xml', function(xml) {
		req = xml;
	}, false);  // async=false so it waits and puts the xml into req
	getFirstChild(getFirstChild(getChild(req.documentElement,1))).childNodes[0].nodeValue = mgid;  //input id
	reqtxt = new XMLSerializer().serializeToString(req.documentElement);
	xhttp.send(reqtxt);
}

File: techniques/JS_PYServer/req.py
import requests
import xml.etree.ElementTree as ET

host = 'http://172.19.254.3/ws'
ns = {'soap':'http://schemas.xmlsoap.org/soap/envelope/'}  # namespaces for path traversal

with requests.Session() as s:
    s.auth=('admin','manager')
    #s.verify = False  # only needed if https with no SSL certificate
    s.headers.update({'Content-Type': 'text/xml;charset=UTF-8', 'SOAPAction': '', 'Connection': 'keep-alive'})
    #print(s.headers)

    with open('/src/XMLReqs/getCPEpartNumber.xml','r') as f:
        body = f.read()

    #r = s.post(host+'/cpeService', data=body)
    r = s.post('http://127.0.0.1:9000', data=body)
    print(r)

    print('-'*60)   
    resp = ET.fromstring(r.text)
    print(resp)
    print(resp[0][0].find('return/factoryPartNumber').text)
    for item in resp[0][0][0]:
        print(item.tag, item.text)

    print('-'*60)
    print('\n'.join(x+': '+y for x,y in r.headers.items()))
